### PR TITLE
fix: cleanup

### PR DESCRIPTION
**Pull request by revolunet**
Add some code documentation

Extract some code in `lib/e2esdk.ts`

### CODE REVIEW

**Review by matmut7 — APPROVED** [review]

File: src/lib/e2esdk.ts [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+};
+
+/**
+ * return encrypted version and metadata for a given `File`
+ * todo: extract to e2esdk
+ */
+export const readAndEncryptFile = async (
+  file: File,
+  encryptionState: EncryptedFormLocalState
+): Promise<{ encryptedFile: File; metadata: Record<string, any> }> =>
```
Comment (franky47) [inline src/lib/e2esdk.ts:116]: **suggestion**: The following would preserve typing of the output as defined in the `@socialgouv/e2esdk-crypto` package:
```suggestion
): ReturnType<typeof encryptFile> =>
```

**Review by franky47 — APPROVED** [review]
LGTM, some type definitions & schemas could be pulled from the crypto package to avoid redundancy.

File: pages/form.tsx [hunk 2]
```diff
@@ -1,119 +1,49 @@
...
+  message: z.string().optional().nullable(),
+  email: z.string().optional().nullable(),
+  color: z.string().optional().nullable(),
   newsletter: z.boolean().optional().default(false),
   alerts: z.boolean().optional().default(false),
-  files: z.array(z.any()),
+  files: z.array(z.any()).optional(),
   filesMetadata: z.record(
     z.string(),
     z.object({
```
[the remark this answers is not in the document]
Comment (revolunet) [reply pages/form.tsx:37]: just like this : `filesMetadata: fileMetadataSchema` ?

File: src/lib/e2esdk.ts (hunk 1 above)
Comment (revolunet) [reply src/lib/e2esdk.ts]: ` Promise<ReturnType<typeof encryptFile>>` ?

**Review by devthejo — APPROVED** [review]

**Pull request merged**